File: tst/Kroki.Tests/Input/Syntax.pas
unit project1;

interface

uses
  cthreads,
  Classes, formy;

type
  TDateRec = record
    Year: Integer;
    Month: (Jan, Feb, Mar, Apr, May, Jun,
            Jul, Aug, Sep, Oct, Nov, Dec);
    Day: 1..31;
  end;

var
  Record1: TDateRec;
  age:integer;
  a:boolean;
  _Answer: string;

implementation

procedure DoDiv;
var
  x, y: integer;
  z: real;
begin
  x := 5; // x = 5
  y := 3; // y = 3
  z := x / y; // Z = 1.666 (value of x divided by y)
  WriteLn(z);
end;

procedure DoIntDiv;
var
  x, y, z: integer;
begin
  x := 5; // x = 5
  y := 3; // y = 3
  z := x div y; // Z = 1 (value of x divided by y)
  Write(z);
end;

procedure DoMod;
var
  x, y, z: integer;
begin
  x := 5; // x = 5
  y := 3; // y = 3
  z := x mod y; // Z = 2 (remainder of x divided by y)
  WriteLn(z);
end;

procedure DoCombined;
var
  w, x, y, z: integer;
begin
  w := 10; // w = 10
  x := 5; // x = 5
  y := 3; // y = 3
  z := (w - x) * y; // Y = 15 (w subtracted by x and then multiplied by y)
  WriteLn(z);
end;

procedure DoOr;
var
  x, y, z: boolean;
begin
  x := true;
  y := false;
  z := y or x; // z = true
  WriteLn(z);
end;

procedure DoAnd;
var
  x, y, z: boolean;
begin
  x := true;
  y := false;
  z := y and x; // z = false
  write(z);
end;

procedure DoNot;
var
  x, y: boolean;
begin
  x := true;
  y := not x; // y = false
  write(y);
end;

procedure DoXor;
var
  x, y, z: boolean;
begin
  x := true;
  y := false;
  z := y xor x; // z = true
    write(z);
end;

procedure DoDiff;
var
  x, y: Integer;
  isDifferent: boolean;
begin
  x := 5;
  y := 10;
  isDifferent := (x <> y); // isDifferent = true
  write(isDifferent);
end;

procedure DoAllEqual1;
var
  x, y, z: boolean;
  areAllEqual: boolean;
begin
  x := false;
  y := false;
  z := true;
  areAllEqual := ((x = y) and (x = z) and (y = z)); // areAllEqual = false
  write(areAllEqual);
end;

procedure DoAllEqual2;
var
  x, y, z: boolean;
  areAllEqual: boolean;
begin
  x := false;
  y := false;
  z := true;
  areAllEqual := ((x = y) and ((x = (z and y)) = z)); // areAllEqual = true
  WriteLn(areAllEqual);
end;

procedure DoPlus;
var
 x, y, z: integer;

begin
 x := 5; // Value of x is 5
 y := 3; // Value of y is 3
 z := x + y; // Z now has the value 8 (value of x plus y)
 WriteLn(z);
end;

procedure DoMinus;
var
  x, y, z: integer;

begin
  x := 5; // x = 5
  y := 3; // y = 3
  z := x - y; // Z = 2 (value of x minus y)
  Write(z);
end;

procedure DoMul;
var
  x, y, z: integer;
begin
  x := 5; // x = 5
  y := 3; // y = 3
  z := x * y; // Z = 15 (value of x times y)
  write(z);
end;

procedure takethis();
var I: integer;
begin
for I := 1 to 10 do
  writeln(I);
end;

procedure DoSomething;
begin
end;

procedure DoSomethingElse;
begin
end;

procedure DoACompletelyDifferentThing;
begin
end;

procedure DoCase(X,a: integer);
begin
  if not X in [1,2] then
     DoACompletelyDifferentThing
  else
  case X of
    1: DoSomething;
    2: DoSomethingElse;
  end;

  case a of
  0: Writeln(0);
  else
    Writeln('else');
    Writeln(a);
    end;

case a of
  0: Writeln(0);
  else begin
    Writeln('else');
    Writeln(a);
  end;
end;

case a of
  0: Writeln(0);
else
  Writeln('else');
  Writeln(a);
end;
end;

procedure DoChar();
var
  a, b: char;
begin
  a := 'H';  // a = "H"
  b := 'i';  // b = "i"
  WriteLn(a);  // Display "H"
  WriteLn(a, b);  // Display "Hi"
  WriteLn(a, b, '!');  // Display "Hi!"
end;

procedure DoValues();
var
  a: Integer;
  b: Real = 2.3;   // Real is normally the same like Single
  c: Single = 4.1334;
  d: Double = 5.45;
  e: Extended = 92922233;
  f: Boolean;
begin
  f := True;
  a := 1;
  f := False;
  writeln(a,b,c,d,e,f);
end;

function DoText : string;
var
  text1: string;
  text:string;
  character1: char;
  firstWord,secondWord: string;

begin
    character1 := 'H';
     text1 := 'ello World!';
      WriteLn(character1, text1);

  text1 := 'I''m John Doe.';

  text := 'Hello World!';
   WriteLn(text[1]); // Only print "H".

   firstWord := 'Hello';
 secondWord := 'World';
 WriteLn(firstWord + ' ' + secondWord + '!');
 DoText := firstWord + ' ' + secondWord + '!';
end;

procedure EnterChar;
var
  number: Integer; // Declare number as an integer
  _Input: string;
  _Char1: char;
begin
  WriteLn('Enter a char!');
  ReadLn(_Input);
  _Char1 := _Input[1];
  number := Ord(_Char1);
  WriteLn('ASCII is: ', number);
end;

procedure EnterName;
var
 _Input, _Linkage: string;
 begin
   WriteLn('Hello, what is your name?');
   ReadLn(_Input);
   _Linkage := 'Hello, ' + _Input;
   WriteLn(_Linkage);
 end;

procedure DoLoops;
var
 i:integer;
begin
 for i := 3 to 12 do
 begin
   writeln(i);
 end;
  for i := 22 downto 3 do
 begin
   WriteLn(i);
 end;
  i:=31;
 while i>=9 do
 begin
   writeln(i);
   i:=i-1;
 end;
 i:=32;
  repeat
   writeln(i);
 until i<13;
end;

procedure DoDisplay;
var
 var1:integer;
begin
 var1:= 12;
 WriteLn (var1);
 ReadLn;
end;

function DoRetrieve:LongInt;
const
  const1 = 12;
var
  var1:integer;
begin
  ReadLn (var1);
  Result := var1 + const1;
end;

begin
  writeln('Hello World');

    with Record1 do
    begin
      Year := 1922;
      Month := Nov;
      Day := 26;
    end;

  WriteLn ('How old are you?');
  ReadLn (age);
  Write ('Allowed to play: ');
  WriteLn ((23 < age) and (age > 52)); // Display "Allowed to play: True" or "Allowed to play: False"

  if a = False then
     WriteLn('a is false')
   else WriteLn('a is true');

     WriteLn('Do you want to order a pizza?');
   ReadLn(_Answer);
   if _Answer = 'Yes' then
     WriteLn('You decided for yes!')
   else WriteLn('Don''t want to have a pizza?');
end.

File: tst/Kroki.Tests/Output/Syntax.cs
using System;
using System.Collections.Generic;
using System.Text;
using Kroki.Runtime;

namespace Kroki.Example
{
    public class TDateRec
    {
    }

    public static class project1
    {
        public static void DoDiv()
        {
            int x = default;
            int y = default;
            float z = default;
            x = 5;
            y = 3;
            z = x / y;
            Console.WriteLine(z);
        }

        public static void DoIntDiv()
        {
            int x = default;
            int y = default;
            int z = default;
            x = 5;
            y = 3;
            z = x / y;
            Console.Write(z);
        }

        public static void DoMod()
        {
            int x = default;
            int y = default;
            int z = default;
            x = 5;
            y = 3;
            z = x % y;
            Console.WriteLine(z);
        }

        public static void DoCombined()
        {
            int w = default;
            int x = default;
            int y = default;
            int z = default;
            w = 10;
            x = 5;
            y = 3;
            z = (w - x) * y;
            Console.WriteLine(z);
        }

        public static void DoOr()
        {
            bool x = default;
            bool y = default;
            bool z = default;
            x = true;
            y = false;
            z = y || x;
            Console.WriteLine(z);
        }

        public static void DoAnd()
        {
            bool x = default;
            bool y = default;
            bool z = default;
            x = true;
            y = false;
            z = y && x;
            Console.Write(z);
        }

        public static void DoNot()
        {
            bool x = default;
            bool y = default;
            x = true;
            y = !x;
            Console.Write(y);
        }

        public static void DoXor()
        {
            bool x = default;
            bool y = default;
            bool z = default;
            x = true;
            y = false;
            z = y ^ x;
            Console.Write(z);
        }

        public static void DoDiff()
        {
            int x = default;
            int y = default;
            bool isDifferent = default;
            x = 5;
            y = 10;
            isDifferent = (x != y);
            Console.Write(isDifferent);
        }

        public static void DoAllEqual1()
        {
            bool x = default;
            bool y = default;
            bool z = default;
            bool areAllEqual = default;
            x = false;
            y = false;
            z = true;
            areAllEqual = ((x == y) && (x == z) && (y == z));
            Console.Write(areAllEqual);
        }

        public static void DoAllEqual2()
        {
            bool x = default;
            bool y = default;
            bool z = default;
            bool areAllEqual = default;
            x = false;
            y = false;
            z = true;
            areAllEqual = ((x == y) && ((x == (z && y)) == z));
            Console.WriteLine(areAllEqual);
        }

        public static void DoPlus()
        {
            int x = default;
            int y = default;
            int z = default;
            x = 5;
            y = 3;
            z = x + y;
            Console.WriteLine(z);
        }

        public static void DoMinus()
        {
            int x = default;
            int y = default;
            int z = default;
            x = 5;
            y = 3;
            z = x - y;
            Console.Write(z);
        }

        public static void DoMul()
        {
            int x = default;
            int y = default;
            int z = default;
            x = 5;
            y = 3;
            z = x * y;
            Console.Write(z);
        }

        public static void takethis()
        {
            int I = default;
            for (I = 1; I <= 10; I++)
            {
                Console.WriteLine(I);
            }
        }

        public static void DoSomething()
        {
        }

        public static void DoSomethingElse()
        {
        }

        public static void DoACompletelyDifferentThing()
        {
        }

        public static void DoCase(int X, int a)
        {
            if (!X.In(new[] { 1, 2 }))
            {
                DoACompletelyDifferentThing();
            }
            else
            {
                switch (X)
                {
                    case 1:
                        DoSomething();
                        break;
                    case 2:
                        DoSomethingElse();
                        break;
                    default:
                        break;
                }
            }

            switch (a)
            {
                case 0:
                    Console.WriteLine(0);
                    break;
                default:
                    Console.WriteLine("else");
                    Console.WriteLine(a);
                    break;
            }

            switch (a)
            {
                case 0:
                    Console.WriteLine(0);
                    break;
                default:
                    Console.WriteLine("else");
                    Console.WriteLine(a);
                    break;
            }

            switch (a)
            {
                case 0:
                    Console.WriteLine(0);
                    break;
                default:
                    Console.WriteLine("else");
                    Console.WriteLine(a);
                    break;
            }
        }

        public static void DoChar()
        {
            char a = default;
            char b = default;
            a = "H";
            b = "i";
            Console.WriteLine(a);
            Console.WriteLine(a, b);
            Console.WriteLine(a, b, "!");
        }

        public static void DoValues()
        {
            int a = default;
            float b = default;
            float c = default;
            double d = default;
            long e = default;
            bool f = default;
            f = true;
            a = 1;
            f = false;
            Console.WriteLine(a, b, c, d, e, f);
        }

        public static string DoText()
        {
            string text1 = default;
            string text = default;
            char character1 = default;
            string firstWord = default;
            string secondWord = default;
            character1 = "H";
            text1 = "ello World!";
            Console.WriteLine(character1, text1);
            text1 = "I''m John Doe.";
            text = "Hello World!";
            Console.WriteLine(text(1));
            firstWord = "Hello";
            secondWord = "World";
            Console.WriteLine(firstWord + " " + secondWord + "!");
            return firstWord + " " + secondWord + "!";
        }

        public static void EnterChar()
        {
            int number = default;
            string _Input = default;
            char _Char1 = default;
            Console.WriteLine("Enter a char!");
            Console.ReadLine(_Input);
            _Char1 = _Input(1);
            number = Ord(_Char1);
            Console.WriteLine("ASCII is: ", number);
        }

        public static void EnterName()
        {
            string _Input = default;
            string _Linkage = default;
            Console.WriteLine("Hello, what is your name?");
            Console.ReadLine(_Input);
            _Linkage = "Hello, " + _Input;
            Console.WriteLine(_Linkage);
        }

        public static void DoLoops()
        {
            int i = default;
            for (i = 3; i <= 12; i++)
            {
                Console.WriteLine(i);
            }

            for (i = 22; i >= 3; i--)
            {
                Console.WriteLine(i);
            }

            i = 31;
            while (i >= 9)
            {
                Console.WriteLine(i);
                i = i - 1;
            }

            i = 32;
            do
            {
                Console.WriteLine(i);
            }
            while (!(i < 13));
        }

        public static void DoDisplay()
        {
            int var1 = default;
            var1 = 12;
            Console.WriteLine(var1);
            Console.ReadLine();
        }

        public static uint DoRetrieve()
        {
            const int const1 = 12;
            int var1 = default;
            Console.ReadLine(var1);
            return var1 + const1;
        }

        private static void Main(string[] args)
        {
            Console.WriteLine("Hello World");
            {
                Record1.Year = 1922;
                Record1.Month = Nov;
                Record1.Day = 26;
            }

            Console.WriteLine("How old are you?");
            Console.ReadLine(age);
            Console.Write("Allowed to play: ");
            Console.WriteLine((23 < age) && (age > 52));
            if (a == false)
            {
                Console.WriteLine("a is false");
            }
            else
            {
                Console.WriteLine("a is true");
            }

            Console.WriteLine("Do you want to order a pizza?");
            Console.ReadLine(_Answer);
            if (_Answer == "Yes")
            {
                Console.WriteLine("You decided for yes!");
            }
            else
            {
                Console.WriteLine("Don''t want to have a pizza?");
            }
        }
    }
}
